searching/double_hasing.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * double hasing.
```

Comment:
Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..